index.js

```diff
@@ -111,13 +115,22 @@ class I18n {
      */
     find(...alternatives) {
         const key = alternatives.shift();
-        const result = resolve(key, this.translations);
+        const result = this.resolve(key);
         const done = typeof result !== 'undefined' || alternatives.length === 0;
 
         return done ? result : this.find(...alternatives);
     }
 
     /**
+     * Perform key lookup on the translation object and memoise the value
+     * @param  {String} key
+     * @return {Any}
+     */
+    resolve(key) {
+        return this[MEMORY][key] = this[MEMORY].hasOwnProperty(key) ? this[MEMORY][key] : resolve(key, this.translations);
```

Comment:
How about:
```
this[MEMORY][key] || resolve(key, this.translations)
```

Reply:
What if the result is `undefined`, `false`, `0`, `''`, `null`?

Reply:
That's still a falsey value and will resolve to the second statement

Reply:
That's right, it will preform the lookup again. If it found *any* value. Falsy values won't be memoised